dendrite/Complement.Dockerfile

```diff
@@ -0,0 +1,7 @@
+# This Dockerfile prepares an image with Complement/Docker pre-installed.
+# This allows users of this image to issue `docker build` commands to build their HS
+# and then run Complement against it.
+FROM golang:1.15-buster
+RUN curl -fsSL https://get.docker.com -o get-docker.sh && sh get-docker.sh
+ADD https://github.com/matrix-org/complement/archive/master.tar.gz .
+RUN tar -xzf master.tar.gz && cd complement-master && go mod download
```

Comment:
I'm confused. why do we need to download complement here as well as in the build steps?

(and if we're building a complement-specific image, it should probably live in the complement repo, not hidden here?)

Reply:
We download Complement here to install dependencies to slightly speed up builds (hence why we just do `go mod download` and don't run Complement itself), so deps are baked into the image. I'm not comfortable making a complement-specific image at this point as I don't know what other requirements may sprout yet, so I'd rather hide it here for this specific instance.

Reply:
I don't really follow. It's *already* a complement-specific image. It may not work fully or be ready for use by others, but I don't see why those problems are solved by keeping the dockerfile here?

Reply:
Because if people see a Dockerfile in the complement repo people think that:
 - it works fully.
 - it's ready to use by others.

when neither are true. I can add appropriate warnings but it seems easier to leave it in the place which required the creation of the Dockerfile in the first place.